### PR TITLE
move znc to role

### PR DESCRIPTION
**Pull request by PurpleNinja225**
**What this PR does / why we need it**:

**Which issue (if any) this PR fixes**:

Fixes #

**Any other useful info**:

### CODE REVIEW

File: roles/znc/tasks/main.yml [hunk 1]
```diff
@@ -3,28 +3,30 @@
...
-- name: ZNC
+- name: Create ZNC Docker Container
   docker_container:
     name: znc
     image: znc
     pull: true
     volumes:
       - "{{ znc_data_directory }}:/znc-data:rw"
       - "/etc/timezone:/etc/timezone:ro"
+    network_mode: "bridge"
```
Comment (davestephens) [inline roles/znc/tasks/main.yml:18]: Not clear why this was added?

Comment (PurpleNinja225) [reply roles/znc/tasks/main.yml:18]: To give confirmation that the container is run on the bridge mode network as apposed to the host network. It also makes it easy for someone to change a containers network mode if they wanted to use a different method, like overlay or macvlan. It's also on all of the other PRs I've moved over to roles and on a few others, such as openhab.

Comment (davestephens) [reply roles/znc/tasks/main.yml:18]: OK, I know this is picky but I'm going to remove these as I merge the PRs (which I am very grateful for!). 

Few reasons for that:
- This is a change to what's currently being done at the moment.  
- Adding options that are set to defaults increases the amount of stuff required to be understood. Docker's default is bridge networking, so people shouldn't be surprised by the network mode. 
- If we really think people need to change it, then it should be set as a parameter people can set through role variables

Comment (PurpleNinja225) [reply roles/znc/tasks/main.yml]: I can add that functionality

**Pull request merged**